llama_index/objects/base_node_mapping.py

```diff
@@ -0,0 +1,68 @@
+"""Base object types."""
+
+from typing import TypeVar, Generic, Sequence, Type, Any
+from abc import abstractmethod
+from llama_index.data_structs.node import Node
+from llama_index.indices.base import BaseIndex
+from llama_index.indices.vector_store.base import VectorStoreIndex
+from llama_index.tools.types import BaseTool
+
+OT = TypeVar("OT")
+
+
+class BaseObjectNodeMapping(Generic[OT]):
```

Comment:
cc @Disiok @logan-markewich i tried to make this interface a bit more general. I wasn't a fan of the base class taking in a list of objects by default. Since there could be cases (e.g. with the SQL table schema), where you can serialize/deserialize to/from Nodes without needing the list beforehand.

Storing a list of Tools helps because we don't have an easy way to serialize/deserialize to/from a Node/Tool.